### PR TITLE
Get correct parameter for entity name

### PR DESCRIPTION
**Pull request by lewistg**
I signed it!

### CODE REVIEW

**Comment by pauldraper** [comment]
It looks one of the lines is too long.

You can run

```sh
scripts/ci/install_closure_deps.sh
scripts/ci/lint_pull_request.sh
```

It will check the files that have changed from master.

**Comment by joeltine** [comment]
What problem is this PR trying to solve?

**Comment by joeltine** [comment]
Also, we have no CLA for lewistg

**Comment by lewistg** [comment]
@joeltine The issue I experienced related to the regular expression used [here] (https://github.com/google/closure-library/pull/545/files#diff-4d487608ccab672d5ee2a1bcef130b44L521). [It contains a capturing group] (https://github.com/google/closure-library/pull/545/files#diff-4d487608ccab672d5ee2a1bcef130b44L274), that will be passed into the callback [here] (https://github.com/google/closure-library/pull/545/files#diff-4d487608ccab672d5ee2a1bcef130b44L522) as the *second* parameter. The first is the full match.

Here is a little example that exposes the bug:

`goog.string.html.htmlSanitize('', function(url){return url;});`

results in

``.

The ampersands were deleted because of the bug.

**Comment by lewistg** [comment]
Also, I signed the CLA, but I am not sure how I can verify my consent. I will try again maybe.

**Comment by joeltine** [comment]
Huh, seems crazy such a blatant bug would exist this long, but I'm not too surprised as there's no tests.

File: third_party/closure/goog/caja/string/html/htmlparser.js [hunk 1]
```diff
@@ -474,11 +474,14 @@ goog.string.html.HtmlParser.prototype.parse = function(handler, htmlText) {
...
  * Decodes an HTML entity.
  *
+ * @param {string} fullEntity The full entity including the leading
+ * '&', the trailing ';', and the content in between.
  * @param {string} name The content between the '&' and the ';'.
  * @return {string} A single unicode code-point as a string.
  * @private
  */
-goog.string.html.HtmlParser.prototype.lookupEntity_ = function(name) {
+goog.string.html.HtmlParser.prototype.lookupEntity_ = function(fullEntity,
```
Comment (joeltine) [inline third_party/closure/goog/caja/string/html/htmlparser.js]: It's a bit odd to have a documented method with a required param that does nothing. I think I'd prefer to leave this with just `name`, and just have the replace in  `unescapeEntities_` call `lookupEntity_` correctly with the right thing.

File: third_party/closure/goog/caja/string/html/htmlparser.js [hunk 2]
```diff
@@ -518,8 +518,10 @@ goog.string.html.HtmlParser.prototype.stripNULs_ = function(s) {
  */
 goog.string.html.HtmlParser.prototype.unescapeEntities_ = function(s) {
   return s.replace(
-      goog.string.html.HtmlParser.ENTITY_RE_,
-      goog.bind(this.lookupEntity_, this));
+      goog.string.html.HtmlParser.ENTITY_RE_, goog.bind(
```
Comment (lewistg) [inline third_party/closure/goog/caja/string/html/htmlparser.js]: @joeltine I force pushed, so it seems your original comment was deleted. But, I agree about the unused parameter.

**Pull request merged**